Rubberduck.Core/UI/CodeExplorer/Commands/ExportCommand.cs

```diff
@@ -100,7 +100,15 @@ public bool PromptFileNameAndExport(QualifiedModuleName qualifiedModule)
                 var component = ProjectsProvider.Component(qualifiedModule);
                 try
                 {
-                    component.Export(dialog.FileName);
+                    if (component.Type == ComponentType.Document)
+                    {
+                        var path = System.IO.Path.GetDirectoryName(dialog.FileName);
+                        component.ExportAsSourceFile(path);
+                    }
+                    else
+                    {
+                        component.Export(dialog.FileName);
+                    }
```

Comment:
I'm not sure that this is the right place to handle the differences for Document components. The implementation of the component wrappers already does the special casing for Document and UserForm modules, we just need to use the `ExportAsSourceFile` entry point instead of the `Export` entry point.

As such the following diff should be sufficient:

```diff
- component.Export(dialog.FileName);
+ component.ExportAsSourceFile(dialog.FileName);
```